### DISCUSSION TITLE
Can I: Lengthen the page size; Get more requests; Query FJC NoS codes?

### DISCUSSION THREAD

**Discussion by MoniqueRon** [freelawproject/courtlistener#3164] [Q&A]
Hi,

1. Is it possible to change the page size of a search query (set to 20 by default). If so, then how?

2. Regarding the limit of API requests, is it possible to make the limit larger? There aren't enough requests allowed in order to get all of the search query results without making the page size larger.

3. Can you please help me understand how to better use the API? For example, I have tried to use the fjc dataset with the natureofsuit + cause + classaction filters to no success. Can you give me an example of how to access filter on this endpoint using python requests please?

**Comment by mlissner** [comment]
Hi and thanks for asking these questions. 

A few responses:

1. No, unfortunately we can't increase the page size. 20 items isn't a *ton*, but remember that each of the items returned could contain the text of a thousand-page PDF, so even 20 sometimes feels like a performance foot-gun. I'm not entirely closed-minded on this, but I think it'd be risky to do to our poor DB.

2. The number of API requests that we give is usually pretty generous for most needs. It's possible you're not doing authentication properly. If you look in these discussions, there should be some info on making sure you're doing it correctly.

    If you find that you are doing it correctly, it might be worth discussing why you need more API access. If this is for a commercial project, it might make sense to enter an API access agreement (we usually like those for commercial projects anyway).

3. Somebody else was recently asking about NOS: https://github.com/freelawproject/courtlistener/discussions/3125. Want to take a look at that answer, and if it helps, great. If not, can you say more about what you've tried and what you're trying to accomplish? One thing to note: We don't regularly update the FJC data. We only do it when one of our clients asks for the work to be done and supports us financially to do so. It's probably a few years old at this point. 

I hope this helps!
